test.py

```diff
@@ -0,0 +1,162 @@
+# pyGene Fusion local testing
+import json
+from pydantic import BaseModel, validator, ValidationError, StrictStr, StrictInt, StrictBool
+from typing import Optional, List, Union
+
+class Gene(BaseModel):
+    type = 'GeneDescriptor'
+    value_id: str
+    label: StrictStr
+
+    @validator('value_id')
+    def is_valid(cls,v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class SequenceRegion(BaseModel):
+    type = 'SequenceLocation'
+    seq_id: str
+    start: StrictInt
+    end: StrictInt
+
+    @validator('seq_id')
+    def is_curie(cls, v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+class ChromosomeRegion(BaseModel):
+    type = 'ChromosomeLocation'
+    chr: str
+    start: str
+    end: str
+
+    @validator('chr')
+    def valid_chr(cls, v):
+        assert v.isalnum(), 'chr must have only alphanumeric characters'
+        return v
+
+    @validator('start', 'end')
+    def valid_loc(cls, v):
+        assert v[0] == 'p' or v[0] == 'q', 'start/end must start with p or q'
+        return v
+
+class GenomicRegion(BaseModel):
+    type = 'computed'
+    description: Optional[str] = None
+    value: Union[SequenceRegion, ChromosomeRegion]
+
+class TranscriptRegion(BaseModel):
+    component_type = 'transcript_segment'
+    transcript: str
+    exon_start: StrictInt
+    exon_start_offset : StrictInt
+    exon_end: StrictInt
+    exon_end_offset : StrictInt
+    gene: Gene
+    component_genomic_region: GenomicRegion
```

Comment:
I think some of these values might be optional? i.e. `exon_start_offset: Optional[StrictInt]`

Reply:
I asked Alex about this - I believe offset should always be present, but can be set to 0.

Reply:
I'm thinking that we should set defaults for these fields then

Reply:
I believe offset values are required, but would equal 0 by default.

Reply:
Yes, 0 default. 👍